protocol/messenger_mailserver.go

```diff
@@ -464,46 +469,48 @@ func (m *Messenger) syncFiltersFrom(filters []*transport.Filter, lastRequest uin
 		m.config.messengerSignalsHandler.HistoryRequestStarted(len(batches))
 	}
 
-	batchKeys := make([]int, 0, len(batches))
-	for k := range batches {
-		batchKeys = append(batchKeys, k)
-	}
-	sort.Ints(batchKeys)
-
 	var batches24h []MailserverBatch
-	keysToIterate := append([]int{}, batchKeys...)
-	for {
-		// For all batches
-		var tmpKeysToIterate []int
-		for _, k := range keysToIterate {
-			batch := batches[k]
-
-			dayBatch := MailserverBatch{
-				To:          batch.To,
-				Cursor:      batch.Cursor,
-				PubsubTopic: batch.PubsubTopic,
-				Topics:      batch.Topics,
-				ChatIDs:     batch.ChatIDs,
-			}
-
-			from := batch.To - 86400
-			if from > batch.From {
-				dayBatch.From = from
-				batches24h = append(batches24h, dayBatch)
+	for pubsubTopic := range batches {
+		batchKeys := make([]int, 0, len(batches[pubsubTopic]))
+		for k := range batches[pubsubTopic] {
+			batchKeys = append(batchKeys, k)
+		}
+		sort.Ints(batchKeys)
+
+		keysToIterate := append([]int{}, batchKeys...)
+		for {
+			// For all batches
+			var tmpKeysToIterate []int
+			for _, k := range keysToIterate {
+				batch := batches[pubsubTopic][k]
+
+				dayBatch := MailserverBatch{
+					To:          batch.To,
+					Cursor:      batch.Cursor,
+					PubsubTopic: batch.PubsubTopic,
+					Topics:      batch.Topics,
+					ChatIDs:     batch.ChatIDs,
+				}
 
-				// Replace og batch with new dates
-				batch.To = from
-				batches[k] = batch
-				tmpKeysToIterate = append(tmpKeysToIterate, k)
-			} else {
-				batches24h = append(batches24h, batch)
+				from := batch.To - 86400
```

Comment:
maybe we can const this number? I take is to allow some drift, but a variable/name explanation would be good